pilot/pkg/model/xds_cache.go

```diff
@@ -30,35 +30,28 @@ import (
 	"istio.io/pkg/monitoring"
 )
 
-func init() {
-	monitoring.MustRegister(xdsCacheReads)
-	monitoring.MustRegister(xdsCacheEvictions)
-	monitoring.MustRegister(xdsCacheSize)
-	monitoring.MustRegister(dependentConfigSize)
+var enableStats = func() bool {
+	return features.EnableXDSCacheMetrics
```

Comment:
no need a func wrapper

Reply:
Sorry. I did not understand. Can you please eloborate?

Reply:
I mean

var enableStats = features.EnableXDSCacheMetrics

Reply:
But RegisterIf takes a func that returns bool?

Reply:
OK, but i doubt it is not necessary in this case.

Reply:
Are you suggesting to rework RegisterIf function?

Reply:
I am not sure, just feel it makes things a little complex with cb